src/odemis/acq/stitching/_registrar.py

```diff
@@ -645,10 +645,13 @@ def _get_shift(self, prev_tile, tile):
             return exp_tile_dist_px, 0
         # Normalized cross correlation (values between -1 and 1)
         ncc = (covar / (st_dev[0] * st_dev[1]))
-        overlap = numpy.abs(
-            numpy.subtract(tile.shape[::-1], numpy.abs(exp_tile_dist_px)))  # tile.shape is YX, so reverse it
-        if numpy.any(numpy.abs(meas_tile_dist_px) > overlap):
-            logging.info(f"Calculated shift {meas_tile_dist_px} is larger than the overlap size {overlap}, "
+        # tile.shape is YX, so reverse it
+        min_overlap = min(numpy.abs(numpy.subtract(tile.shape[::-1], numpy.abs(exp_tile_dist_px))))
+        # If the measured shift in x or y is higher than the minimum overlap, fall back to the expected shift.
+        # Use the minimum overlap because the overlapping parts are often elongated, even for elongated areas we do
+        # not expect a shift higher than the smallest side of the overlapping area.
```

Comment:
I agree that on the "non-shifted" dimension (which is typically the most elongated one), the shift should not be very large. Using along this dimension the same value as for the "shifted dimension" is fine, but I think the comment should be more clear that it's just a "heuristic". You could even be more picky and say N% of the "shifted dimension".